### PR TITLE
Updated functions documentation

### PR DESCRIPTION
**Pull request by matthewryanwells**
### Description
There were a few mistakes in the function documentation that I have fixed.
- Added missing arithmetic operators
- Added missing data types
- Added missing functions
- Fixed incorrect function usage

I also made a few general improvements to the readability of the code.

### Issues Resolved
This PR is not associated with any issues

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _search-plugins/sql/functions.md [hunk 1]
```diff
@@ -19,12 +19,12 @@ The SQL plugin supports the following functions.
 Function | Specification | Example
 :--- | :--- | :---
 abs | `abs(number T) -> T` | `SELECT abs(0.5) FROM my-index LIMIT 1`
```
Comment (acarbonetto) [inline _search-plugins/sql/functions.md:21]: @matthewryanwells the `FROM my-index LIMIT 1` throughout the file can be removed.  The SQL plug no longer requires that a `FROM` table is included, and the statement `SELECT abs(0.5)` is now valid.

**Review by acarbonetto — APPROVED** [review]
thanks!

**Review by Yury-Fridlyand — APPROVED** [review]

**Comment by kolchfa-aws** [comment]
Thank you, @matthewryanwells, @acarbonetto, and @Yury-Fridlyand! This needs to be backported to all versions 1.3 and up, correct?

File: _search-plugins/sql/functions.md [hunk 2]
```diff
@@ -18,166 +18,170 @@ The SQL plugin supports the following common functions shared across the SQL and
...
+| timestampadd       | `timestampadd(interval, integer, date) -> date)`                                       | `SELECT timestampadd(DAY, 17, datetime('2000-01-01 00:00:00'))`                     |
+| timestampdiff      | `timestampdiff(interval, date, date) -> integer`                                       | `SELECT timestampdiff(YEAR, '1997-01-01 00:00:00, '2001-03-06 00:00:00')`           |
+| time_format        | `time_format(date, string) -> string`                                                  | `SELECT time_format('1998-01-31 13:14:15.012345', '%f %H %h %I %i %p %r %S %s %T')` |
+| time_to_sec        | `time_to_sec(time) -> long`                                                            | `SELECT time_to_sec(time '22:23:00')`                                               |
+| to_days            | `to_days(date) -> long`                                                                | `SELECT to_days(date '2008-10-07')`                                                 |
+| to_seconds         | `to_seconds(date) -> integer`                                                          | `SELECT to_seconds(date('2008-10-07')`                                              |
```
Comment (acarbonetto) [inline _search-plugins/sql/functions.md:141]:
```suggestion
| to_seconds         | `to_seconds(date) -> integer`                                                          | `SELECT to_seconds(date('2008-10-07'))`                                              |
```

File: _search-plugins/sql/functions.md [hunk 3]
```diff
@@ -18,166 +18,170 @@ The SQL plugin supports the following common functions shared across the SQL and
...
 ## Advanced
 
-Function | Specification | Example
-:--- | :--- | :---
-if | `if(boolean, es_type, es_type) -> es_type` | `SELECT if(false, 0, 1) FROM my-index LIMIT 1`, `SELECT if(true, 0, 1) FROM my-index LIMIT 1`
-ifnull | `ifnull(es_type, es_type) -> es_type` | `SELECT ifnull('hello', 1) FROM my-index LIMIT 1`, `SELECT ifnull(null, 1) FROM my-index LIMIT 1`
-isnull | `isnull(es_type) -> integer` | `SELECT isnull(null) FROM my-index LIMIT 1`, `SELECT isnull(1) FROM my-index LIMIT 1`
+| Function | Specification                              | Example                                               |
+|:---------|:-------------------------------------------|:------------------------------------------------------|
+| if       | `if(boolean, es_type, es_type) -> es_type` | `SELECT if(false, 0, 1)`, `SELECT if(true, 0, 1)`     |
```
Comment (acarbonetto) [inline _search-plugins/sql/functions.md:182]: can you double-check that these work please? 

- [ ] test

File: _search-plugins/sql/functions.md [hunk 4]
```diff
@@ -18,166 +18,170 @@ The SQL plugin supports the following common functions shared across the SQL and
...
+| day_of_year\*      | `day_of_year(date) -> integer`                                                         | `SELECT day_of_year(date('2020-08-26'))`                                            |
+| extract\*          | `extract(part FROM date) -> integer`                                                   | `SELECT extract(MONTH FROM datetime('2020-08-26 10:11:12'))`                        |
+| from_days          | `from_days(N) -> integer`                                                              | `SELECT from_days(733687)`                                                          |
+| from_unixtime      | `from_unixtime(N) -> date`                                                             | `SELECT from_unixtime(1220249547)`                                                  |
+| get_format         | `get_format(PART, string) -> string`                                                   | `SELECT get_format(DATE, 'USA')`                                                    |
+| hour               | `hour(time) -> integer`                                                                | `SELECT hour((time '01:02:03'))`                                                    |
```
Comment (acarbonetto) [inline _search-plugins/sql/functions.md:109]:
```suggestion
| hour               | `hour(time) -> integer`                                                                | `SELECT hour(time '01:02:03')`                                                    |
```

File: _search-plugins/sql/functions.md [hunk 5]
```diff
@@ -18,166 +18,170 @@ The SQL plugin supports the following common functions shared across the SQL and
...
+| hour_of_day\*      | `hour_of_day(time) -> integer`                                                         | `SELECT hour_of_day((time '01:02:03'))`                                             |
```
Comment (acarbonetto) [inline _search-plugins/sql/functions.md:110]:
```suggestion
| hour_of_day\*      | `hour_of_day(time) -> integer`                                                         | `SELECT hour_of_day(time '01:02:03')`                                             |
```

File: _search-plugins/sql/functions.md [hunk 6]
```diff
@@ -18,166 +18,170 @@ The SQL plugin supports the following common functions shared across the SQL and
...
+| localtime          | `localtime() -> date`                                                                  | `SELECT localtime()`                                                                |
+| localtimestamp     | `localtimestamp() -> date`                                                             | `SELECT localtimestamp()`                                                           |
+| makedate           | `makedate(double, double) -> date`                                                     | `SELECT makedate(1945, 5.9)`                                                        |
+| maketime           | `maketime(integer, integer, integer) -> date`                                          | `SELECT maketime(1, 2, 3)`                                                          |
+| microsecond        | `microsecond(expr) -> integer`                                                         | `SELECT microsecond((time '01:02:03.123456'))`                                      |
+| minute             | `minute(expr) -> integer`                                                              | `SELECT minute((time '01:02:03'))`                                                  |
```
Comment (acarbonetto) [inline _search-plugins/sql/functions.md:117]:
```suggestion
| minute             | `minute(expr) -> integer`                                                              | `SELECT minute(time '01:02:03')`                                                  |
```

File: _search-plugins/sql/functions.md [hunk 7]
```diff
@@ -18,166 +18,170 @@ The SQL plugin supports the following common functions shared across the SQL and
...
+| minute_of_day\*    | `minute_of_day(expr) -> integer`                                                       | `SELECT minute_of_day((time '01:02:03'))`                                           |
```
Comment (acarbonetto) [inline _search-plugins/sql/functions.md:118]:
```suggestion
| minute_of_day\*    | `minute_of_day(expr) -> integer`                                                       | `SELECT minute_of_day(time '01:02:03')`                                           |
```

File: _search-plugins/sql/functions.md [hunk 8]
```diff
@@ -18,166 +18,170 @@ The SQL plugin supports the following common functions shared across the SQL and
...
+| minute_of_hour\*   | `minute_of_hour(expr) -> integer`                                                      | `SELECT minute_of_hour((time '01:02:03'))`                                          |
```
Comment (acarbonetto) [inline _search-plugins/sql/functions.md:119]:
```suggestion
| minute_of_hour\*   | `minute_of_hour(expr) -> integer`                                                      | `SELECT minute_of_hour(time '01:02:03')`                                          |
```

File: _search-plugins/sql/functions.md [hunk 9]
```diff
@@ -18,166 +18,170 @@ The SQL plugin supports the following common functions shared across the SQL and
...
+| monthname          | `monthname(date) -> string`                                                            | `SELECT monthname(date('2020-08-26'))`                                              |
+| now                | `now() -> date`                                                                        | `SELECT now()`                                                                      |
+| period_add         | `period_add(integer, integer)`                                                         | `SELECT period_add(200801, 2)`                                                      |
+| period_diff        | `period_diff(integer, integer)`                                                        | `SELECT period_diff(200802, 200703)`                                                |
+| quarter            | `quarter(date) -> integer`                                                             | `SELECT quarter(date('2020-08-26'))`                                                |
+| second             | `second(time) -> integer`                                                              | `SELECT second((time '01:02:03'))`                                                  |
```
Comment (acarbonetto) [inline _search-plugins/sql/functions.md:127]:
```suggestion
| second             | `second(time) -> integer`                                                              | `SELECT second(time '01:02:03')`                                                  |
```

File: _search-plugins/sql/functions.md [hunk 10]
```diff
@@ -18,166 +18,170 @@ The SQL plugin supports the following common functions shared across the SQL and
...
+| second_of_minute\* | `second_of_minute(time) -> integer`                                                    | `SELECT second_of_minute((time '01:02:03'))`                                        |
```
Comment (acarbonetto) [inline _search-plugins/sql/functions.md:128]:
```suggestion
| second_of_minute\* | `second_of_minute(time) -> integer`                                                    | `SELECT second_of_minute(time '01:02:03')`                                        |
```

**Review by acarbonetto — APPROVED** [review]
Looks good, please test the boolean functions and make sure that they are supported in V2

File: _search-plugins/sql/functions.md (hunk 3 above)
Comment (matthewryanwells) [reply _search-plugins/sql/functions.md]: I checked and there was one incorrect example, fixed now.

I am also realizing the it says es_type which I am assuming stands for elastic search type, I can change that to os_type

File: _search-plugins/sql/functions.md (hunk 10 above)
Comment (matthewryanwells) [reply _search-plugins/sql/functions.md]: fixed in https://github.com/opensearch-project/documentation-website/pull/4232/commits/9442d3b1e8ec65a7c0e9b166073cf2b8281553d1

File: _search-plugins/sql/functions.md (hunk 9 above)
Comment (matthewryanwells) [reply _search-plugins/sql/functions.md]: fixed in https://github.com/opensearch-project/documentation-website/pull/4232/commits/9442d3b1e8ec65a7c0e9b166073cf2b8281553d1

File: _search-plugins/sql/functions.md (hunk 7 above)
Comment (matthewryanwells) [reply _search-plugins/sql/functions.md]: fixed in https://github.com/opensearch-project/documentation-website/pull/4232/commits/9442d3b1e8ec65a7c0e9b166073cf2b8281553d1

File: _search-plugins/sql/functions.md (hunk 5 above)
Comment (matthewryanwells) [reply _search-plugins/sql/functions.md]: fixed in https://github.com/opensearch-project/documentation-website/pull/4232/commits/9442d3b1e8ec65a7c0e9b166073cf2b8281553d1

File: _search-plugins/sql/functions.md (hunk 4 above)
Comment (matthewryanwells) [reply _search-plugins/sql/functions.md]: fixed in https://github.com/opensearch-project/documentation-website/pull/4232/commits/9442d3b1e8ec65a7c0e9b166073cf2b8281553d1

File: _search-plugins/sql/functions.md [hunk 11]
```diff
@@ -18,166 +18,170 @@ The SQL plugin supports the following common functions shared across the SQL and
...
+| dayname            | `dayname(date) -> string`                                                              | `SELECT dayname(date('2020-08-26'))`                                                |
+| dayofmonth         | `dayofmonth(date) -> integer`                                                          | `SELECT dayofmonth(date('2020-08-26'))`                                             |
+| dayofweek          | `dayofweek(date) -> integer`                                                           | `SELECT dayofweek(date('2020-08-26'))`                                              |
+| dayofyear          | `dayofyear(date) -> integer`                                                           | `SELECT dayofyear(date('2020-08-26'))`                                              |
+| dayofweek          | `dayofweek(date) -> integer`                                                           | `SELECT dayofweek(date('2020-08-26'))`                                              |
+| day_of_month\*     | `day_of_month(date) -> integer`                                                        | `SELECT day_of_month(('2020-08-26'))`                                               |
```
[the remark this answers is not in the document]
Comment (matthewryanwells) [reply _search-plugins/sql/functions.md]: fixed in https://github.com/opensearch-project/documentation-website/pull/4232/commits/9442d3b1e8ec65a7c0e9b166073cf2b8281553d1

File: _search-plugins/sql/functions.md (hunk 2 above)
Comment (matthewryanwells) [reply _search-plugins/sql/functions.md]: fixed in https://github.com/opensearch-project/documentation-website/pull/4232/commits/9442d3b1e8ec65a7c0e9b166073cf2b8281553d1

**Comment by matthewryanwells** [comment]
@kolchfa-aws the documentation changes made by me, @acarbonetto, and @Yury-Fridlyand are for version 2.8 and above.

**Comment by kolchfa-aws** [comment]
Thank you, @matthewryanwells! Do you mind surrounding all the function names in the first column with tic marks to make the PR pass the linter?

**Comment by matthewryanwells** [comment]
@kolchfa-aws I have added the tics to the function names, and also went through each and every query example works as written

**Review by kolchfa-aws — APPROVED** [review]
Thanks so much for your contribution, @matthewryanwells!

**Pull request merged**